content/en/platform/corda/5.0/developing-applications/api/ledger/utxo-ledger/vault-queries.md

```diff
@@ -0,0 +1,561 @@
+                .collect(new DummyCustomQueryCollector())
+                .register();
+    }
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{< note >}}
+The collector always needs to be the last one in the chain as all previous filtering and mapping functionality will be applied before collecting.
+{{< /note >}}
+
+### Executing a Vault-Named Query
+
+To execute a query use `UtxoLedgerService`. This can be injected to a flow via `@CordaInject`.
+To instantiate a query call the following:
+
+{{< tabs name="tabs-8" >}}
+{{% tab name="Kotlin" %}}
+
+```kotlin
+utxoLedgerService.query("DUMMY_CUSTOM_QUERY", Int::class.java)
+```
+
+{{% /tab %}}
+{{% tab name="Java" %}}
+
+```java
+utxoLedgerService.query("DUMMY_CUSTOM_QUERY", Integer.class)
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+Provide the name of the query (in this case `DUMMY_CUSTOM_QUERY`) and the return type. Since the result set is collected into an integer in the complex query example use `Int` (or `Integer` in Java).
+
+Before executing define the following:
+
+* Which index the result set should start (`setOffset`), default 0.
+* How many results should the query return (`setLimit`), default Int.MAX (2,147,483,647).
+* Define named parameters that are in the query and the actual value for them. {{< note >}} All parameters must be defined otherwise the execution will fail. (`setParameter` or `setParameters`) {{</ note >}}
+* Each state in the database has a timestamp value for when it was inserted. Set an * upper limit to only return states that were inserted before a given time. (`setTimestampLimit`)
+
+In this case it would look like this:
+
+{{< tabs name="tabs-9" >}}
+{{% tab name="Kotlin" %}}
+
+```kotlin
+    val resultSet = utxoLedgerService.query("DUMMY_CUSTOM_QUERY", Int::class.java) // instantiate the query
+                .setOffset(0) // Start from the beginning
+                .setLimit(1000) // Only return 1000 records
+                .setParameter("testField", "dummy") // Set the parameter to a dummy value
+                .setCreatedTimestampLimit(Instant.now()) // Set the timestamp limit to the current time
+                .execute() // execute the query
+```
+
+{{% /tab %}}
+{{% tab name="Java" %}}
+
+```java
+    PagedQuery.ResultSet<Integer> resultSet = utxoLedgerService.query("DUMMY_CUSTOM_QUERY", Integer.class) // instantiate the query
+                    .setOffset(0) // Start from the beginning
+                    .setLimit(1000) // Only return 1000 records
+                    .setParameter("testField", "dummy") // Set the parameter to a dummy value
+                    .setCreatedTimestampLimit(Instant.now()) // Set the timestamp limit to the current time
+                    .execute(); // execute the query
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{< note >}}
+A dummy value is assigned for the `testField` parameter in this query but it can be replaced. There is only one parameter in this example query which is `:testField`.
```

Comment:
```suggestion
A dummy value is assigned for the `testField` parameter in this query, but it can be replaced. There is only one parameter in this example query which is `:testField`.
```